src/udev/udev-builtin-input_id.c

```diff
@@ -240,13 +241,18 @@ static bool test_pointers(struct udev_device *dev,
                         is_touchscreen = true;
         }
 
-        if (has_rel_coordinates && has_mouse_button)
-                is_mouse = true;
+        if (has_mouse_button)
+                if (has_rel_coordinates)
+                        is_mouse = true;
+                if (!has_abs_coordinates) /* exclude vmmouse and joysticks */
+                        is_pad = true;
 
         if (is_pointing_stick)
                 udev_builtin_add_property(dev, test, "ID_INPUT_POINTINGSTICK", "1");
         if (is_mouse)
                 udev_builtin_add_property(dev, test, "ID_INPUT_MOUSE", "1");
+        if (is_pad)
+                udev_builtin_add_property(dev, test, "ID_INPUT_MOUSE", "1");
```

Comment:
TBH I don't understand this. The net effect of this patch  should be exactly the same, except for the case where there is a device without relative nor absolute axes: your patch would currently label it a a mouse (which sounds completely wrong), and with dropping the `!` above it would behave it as right now (which then makes little sense). And the `is_<device>` really ought to coincide with the `ID_INPUT_<DEVICE>` variables, please don't  mix up pads and mouses.